support-frontend/app/controllers/DigitalSubscriptionController.scala

```diff
@@ -184,3 +173,28 @@ class DigitalSubscriptionController(
   }
 
 }
+
+class LandingCopyProvider(
+  priceSummaryServiceProvider: PriceSummaryServiceProvider,
+  promotionServiceProvider: PromotionServiceProvider,
+  stage: Stage
+) {
+
+  type PromoDetails = (ProductPrices, Option[PromotionCopy])
+
+  def promoDetails(queryPromos: List[String], countryCode: String, orderIsAGift: Boolean): PromoDetails = {
+    val promoCodes: List[PromoCode] = queryPromos ++ DefaultPromotions.DigitalSubscription.all
+
+    val country = (for {
+      countryGroup <- CountryGroup.byId(countryCode)
+      country <- countryGroup.countries.headOption
+    } yield country).getOrElse(UK)
+    val promoCode = queryPromos.headOption.getOrElse(DefaultPromotions.DigitalSubscription.landing)
+    val promotionCopy = ProductPromotionCopy(promotionServiceProvider.forUser(false), stage)
+      .getCopyForPromoCode(promoCode, DigitalPack, country)
+    val readerType = if (orderIsAGift) Gift else Direct
```

Comment:
I'm not sure it adds much putting the price stuff in here - it complicates the return type and then we immediately extract the tuple in the calling method. Why not just deal with the copy in here and leave prices outside?

Reply:
good question.  The above method is perilously close to the 50 lines limit so I was a bit enthusiastic.  However you are right that they are mostly independent so they shouldn't be mixed here either.
I was half hoping the extracted method could be used in Paper and Weekly in the next PR.
I'll rejig it in this PR along the lines you suggest.